.github/workflows/release.yml

```diff
@@ -268,4 +268,55 @@ jobs:
       uses: actions/upload-artifact@v2
       with:
         name: macos-signed-installer
-        path: signed/*.pkg
\ No newline at end of file
+        path: signed/*.pkg
+
+# ================================
+#              Windows
+# ================================
+  build-and-lay-out-windows:
+    name: Build and lay out Windows
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # Indicate full history so Nerdbank.GitVersioning works.
+
+    - name: Set up dotnet
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.201
+
+    # Install Nerdbank.GitVersioning
+    - uses: dotnet/nbgv@master
+      with:
+        setCommonVars: true
+    
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Build
+      run: |
+        dotnet build --configuration=WindowsRelease
+
+    - name: Run Windows unit tests
+      run: |
+        dotnet test --configuration=WindowsRelease
+    
+    - name: Lay out
+      shell: pwsh
+      run: |
+        mkdir windows-exes,windows-payload,windows-symbols
+        Move-Item -Path out\windows\Payload.Windows\bin\Release\net472\win-x86\* -Destination windows-payload
+        Move-Item -Path windows-payload\*.pdb -Destination windows-symbols
+        Compress-Archive -Path windows-payload/* ./win-x86-$env:GitBuildVersionSimple.zip
+        Compress-Archive -Path windows-symbols/* ./symbols-win-x86-$env:GitBuildVersionSimple.zip
+        Move-Item -Path out\windows\Installer.Windows\bin\Release\net472\*.exe -Destination windows-exes
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: windows-artifacts
+        path: |
+          windows-exes
+          win-x86-$env:GitBuildVersionSimple.zip
+          symbols-win-x86-$env:GitBuildVersionSimple.zip
```

Comment:
Unfortunately we're going to need to change how our Windows builds work to make it work with the new ESRP signing flow.

Today with the MicroBuild-based build, the `Payload.Windows` project does the code-signing of the payload binaries (`.exe`s and `.dll`s), and then the `Installer.Windows` project takes the _signed_ outputs of the `Payload.Windows` project and creates an unsigned Inno Setup installer, which then is signed via MicroBuild.

We need to split out the creation of the payload and installer in to separately runnable parts in the MSBuild projects, much like we do in macOS with `build.sh` internally calling `layout.sh && pack.sh && dist.sh` when run locally, but in our release workflows we call `layout.sh` and friends separately (because we need to submit to ESRP in between).

Reply:
This is the one thing I haven't yet addressed with my latest changes. I think it would be a good idea for us to pair on this so I'm certain I'm implementing this correctly. Grabbing some time on your calendar.

Reply:
This is the one thing I haven't yet addressed with my latest changes. I think it would be a good idea for us to pair on this so I'm certain I'm implementing this correctly. Grabbing some time on your calendar.

Reply:
I think this is taken care of, although I'm happy to modify if you have suggestions to improve the approach.